tests/jobs.py

```diff
@@ -358,6 +358,31 @@ def test_dagrun_root_fail(self):
             },
             dagrun_state=State.FAILED)
 
+    def test_dagrun_root_fail_unfinished(self):
+        """
+        DagRuns with one unfinished and one failed root task -> RUNNING
+        """
+        # Run both the failed and successful tasks
+        scheduler = SchedulerJob(**self.default_scheduler_args)
+        dag_id = 'test_dagrun_states_root_fail_unfinished'
+        dag = self.dagbag.get_dag(dag_id)
+        dag.clear()
+        dr = scheduler.create_dag_run(dag)
+        try:
+            dag.run(start_date=dr.execution_date, end_date=dr.execution_date)
+        except AirflowException:  # Expect an exception since there is a failed task
+            pass
+
+        # Mark the successful task as never having run since we want to see if the
+        # dagrun will be in a running state despite haveing an unfinished task.
+        session = settings.Session()
+        ti = session.query(TI).filter(TI.dag_id==dag.dag_id,
```

Comment:
Nit: use one() to make sure we can not have indeterministic outcomes here. Maybe smarter to get the task_instance from the dagrun instead? Then you know you will get the right one, as you are also not querying on primary key.